pkg/kubernetes/deploy.js

```diff
@@ -0,0 +1,373 @@
+            has_errors = true;
+        }
+        if (ns.trim() === '' || ns.trim() === 'Enter Namespace Here') {
+            $('#deploy-app-namespace-field-note').show();
+            $('#deploy-app-namespace-field-custom').parent().addClass('has-error');
+            has_errors = true;
+        }
+        if (!has_errors)
+            appdeployer.manager.deploy_app(ns, jsondata);
+    }
+
+    function deploy_dialog_remove_errors() {
+        $('#deploy-app-manifest-file-note').hide();
+        $('#deploy-app-manifest-file-note-details').hide();
+        $('#deploy-app-namespace-field-note').hide();
+        $('#deploy-app-namespace-field-note-details').hide();
+        $('#deploy-app-general-error').hide();
+        $('#deploy-app-manifest-file-empty').hide();
+        $('#deploy-app-namespace-field-custom-empty').hide();
+        $('#deploy-app-namespace-field-custom').parent().removeClass('has-error');
+        $('#deploy-app-manifest-file').parent().removeClass('has-error');
+
+    }
+
+    function pre_init() {
+        //alert("pre_init")
+        var firstTime = true;
+        var dlg = $('#deploy-app-dialog');
+        var btn = $('#deploy-app-start');
+        var manifest_file = $('#deploy-app-manifest-file');
+        var manifest_file_note = $('#deploy-app-manifest-file-note');
+        var manifest_file_details = $("#deploy-app-manifest-file-note-details");
+        var ns_selector = $('#deploy-app-namespace-field');
+        appdeployer.jsondata = "";
+        var text = "";
+
+        btn.on('click', function() {
+            deploy_app();
+        });
+
+        dlg.on('show.bs.modal', function() {
+            //alert("show.bs.models");
+            if (firstTime) {
+                var optionls = [];
+                var nslist = get_kentities("namespaces");
+                for (var i = 0; i < nslist.length; i++) {
+                    optionls.push('<option translatable="yes" value="' + nslist[i] + '">' + nslist[i] + '</option>');
+                }
+                optionls.push('<option translatable="yes" value="Custom Namespace">Custom Namespace</option>');
+                var optionlshtml = optionls.join('');
+                ns_selector.prepend(optionlshtml);
+                ns_selector.selectpicker('refresh');
+                firstTime = false;
+            }
+            manifest_file.val("");
+            deploy_dialog_remove_errors();
+        });
+
+
+        dlg.on('keypress', function(e) {
+            if (e.keyCode === 13)
+                btn.trigger('click');
+        });
+
+        manifest_file.on('change', function() {
+            //alert("manifest_file")
+
+            manifest_file_note.hide();
+            manifest_file_details.hide();
+            manifest_file.parent().removeClass('has-error');
+
+            var files, file, reader;
+            files = manifest_file[0].files;
+            if (files.length != 1) {
+                text = "No json File was selected.Please select a json file. ";
```

Comment:
This text is not translatable. Need to wrap this and all other javascript translatable strings in _("").

In addition, there seem to be typos in this string. Suggest:

```
No metadata file was selected. Please select a Kubernetes metadata file."
```